src/core/controllers/base.go

```diff
@@ -54,10 +57,43 @@ type messageDetail struct {
 	UUID string
 }
 
+func redirectForOIDC(ctx context.Context, username string) bool {
+	am, _ := ctx.Value(filter.AuthModeKey).(string)
+	if am != common.OIDCAuth {
+		return false
+	}
+	u, err := dao.GetUser(models.User{Username: username})
+	if err != nil {
+		log.Warningf("Failed to get user by name: %s, error: %v", username, err)
+	}
```

Comment:
return ?

Reply:
+1

Reply:
It's not necessary because if u is nil it returns `true` so when error happens the redirection will happen.